test/e2e/framework.go

```diff
@@ -1529,7 +1536,243 @@ func (data *TestData) Cleanup(namespaces []string) {
+	if err != nil {
+		return fmt.Errorf("copyCovFolder: error creating absolute file path: %v", err)
+	}
+	cmd := exec.Command("docker", "cp", nodeName+":"+"/var/log/"+covPrefix+"-coverage/.", covDirAbs);
+	if err := cmd.Run(); err != nil {
+		if !strings.Contains(err.Error(), "exit status 1") {
+			return fmt.Errorf("error while running docker cp command[%v] from node: %s: %v", cmd, nodeName, err)
+		}
+	}
+	return nil
+}
+
+func (data *TestData) killProcessesAndCollectCovFiles(namespace, podName, containerName, processName, covDir, covPrefix string) error {
+	cmds := []string{"pgrep", "-f", processName}
+	stdout, stderr, err := data.RunCommandFromPod(namespace, podName, containerName, cmds)
+	if err != nil {
+		return fmt.Errorf("error when getting pid of '%s', stderr: <%v>, err: <%v>", processName, stderr, err)
+	}
+	cmds = []string{"kill", "-SIGINT", strings.TrimSpace(stdout)}
+	log.Infof("Sending SIGINT to '%s' with cmd 'kill -SIGINT %s'", processName, strings.TrimSpace(stdout))
+	fmt.Println("sending SIGINT")
+	_, stderr, err = data.RunCommandFromPod(namespace, podName, containerName, cmds)
+	if err != nil {
+		return fmt.Errorf("error when sending SIGINT signal to '%s', stderr: <%v>, err: <%v>", processName, stderr, err)
+	}
+	log.Infof("Copying coverage files from worker nodes kind-worker and kind-worker2")
+	if err = data.copyCovFolder("kind-worker", covDir, covPrefix); err != nil {
+		return err
+	}
+	if err = data.copyCovFolder("kind-worker2", covDir, covPrefix); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (data *TestData) copyCovFilesFromPods(covDir string, nodeName string) error {
+	listOptions := metav1.ListOptions{
+		LabelSelector: "",
+	}
+	if nodeName != "all" {
+		listOptions.FieldSelector = fmt.Sprintf("spec.nodeName=%s", nodeName)
+	}
+
+	pods, err := data.clientset.CoreV1().Pods(flowVisibilityNamespace).List(context.TODO(), listOptions)
+	if err != nil {
+		return fmt.Errorf("failed to list flow-visibility pods: %v", err)
+	}
+	for _, pod := range pods.Items {
+		podName := pod.Name
+		if strings.Contains(podName, "chi-clickhouse-clickhouse") {
+			err = data.killProcessesAndCollectCovFiles("flow-visibility", pod.Name, clickHouseMonitorContName, clickHouseMonitorContName, covDir+"/"+clickHouseMonitorCovFolder, "cm")
+		} else if strings.Contains(podName, "theia-manager") {
+			err = data.killProcessesAndCollectCovFiles("flow-visibility", pod.Name, theiaManagerContName, theiaManagerContName, covDir+"/"+theiaManagerCovFolder, "tm")
+		} else {
+			fmt.Println("no coverage files to look for")
+		}
+		if err != nil {
+			fmt.Println(fmt.Sprintf("copyCovFilesFromPods: error copying node files: %v", err))
+			return fmt.Errorf("error when copying coverage files from pods: copy pod files out, error:%v", err)
+		}
+	}
+	return nil
+}
+
 func flowVisibilityCleanup(tb testing.TB, data *TestData, config FlowVisibilitySetUpConfig) {
+	// TODO potentially check for files here
+	data.copyCovFilesFromPods(".coverage", "all")
```

Comment:
why are we only copying the coverage from flow visibility tests? what about other tests?

Reply:
added SIGINT handling to teardownFlowVisibility(), and added a test that only runs after the rests of the tests, copies all coverage files generated by all tests, deletes them off of the node, and finally cleans up local files after go tool covdata is used